tests/components/layout/test_cond.py

```diff
@@ -0,0 +1,37 @@
+import pytest
+
+import pynecone as pc
+
+
+@pytest.fixture
+def cond_state(request):
+    class CondState(pc.State):
+        value: request.param["value_type"] = request.param["value"]
+
+    return CondState
+
+
+@pytest.mark.parametrize(
+    "cond_state",
+    [
+        pytest.param({"value_type": bool, "value": True}),
+        pytest.param({"value_type": int, "value": 0}),
+        pytest.param({"value_type": str, "value": "true"}),
+    ],
+    indirect=True,
+)
+def test_validate_cond(cond_state: pc.Var):
+    """Test if cond can be a pc.Val with any values"""
+    cond_component = pc.cond(
+        cond_state.value,
+        pc.text("cond is True"),
+        pc.text("cond is False"),
+    )
```

Comment:
This may because we do some magic to create the `pc.` functions. In the tests maybe we can try to import the components directly:

```python
from pynecone.components.typography.text import Text
from pynecone.components.layout.cond import Cond
```

Then call the create method explicitly:

```python
cond_component = Cond.create(
         cond_state.value,
         Text.create("cond is True"),
         Text.create("cond is False"),
     )
```

Reply:
Thanks. It seems to fix it.

> This may because we do some magic to create the pc. functions.

I am curious how we do this? e.g. I don't see any `def cond():` for `pc.cond`

Reply:
[This is the lin](https://github.com/pynecone-io/pynecone/blob/main/pynecone/components/__init__.py#L20)e that does the magic: we create `pc.` functions that map to each component's `create` method.